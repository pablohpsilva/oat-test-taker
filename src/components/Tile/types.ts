export default interface ITile {
    userid: string
    firstname: string
    lastname: string
    title?: string
    picture?: string
}