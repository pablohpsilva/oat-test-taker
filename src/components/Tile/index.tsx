import React, { FC } from 'react'
import { generatePath } from 'react-router'
import { Link } from 'react-router-dom'
import styled from 'styled-components/macro'

import { Avatar } from '@/components'
import { USER } from '@/config/URL'
import ITile from './types'

const StyledWrapper = styled.div`
    border: var(--s) solid var(--color-dark-1-30);
    border-radius: calc(3 * var(--s));
    background: var(--color-light-0);
    padding: var(--dim-gutter);
    display: flex;
    justify-content: center;
`

const Grid = styled.div`
    display: grid;
    grid-template-columns: auto 10fr;
    grid-template-rows: 1fr;
    grid-gap: var(--dim-gutter);
    width: 100%;
    align-items: center;
`

const ContentGrid = styled.div`
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: auto auto;
    grid-gap: var(--dim-gutter);
    width: 100%;
    align-items: center;
`

const Tile: FC<ITile> = ({ userid, email, title, lastname, firstname, picture }) => {
    return (
        <Link to={generatePath(USER, { userid })}>
            <StyledWrapper>
                <Grid>
                    <Avatar picture={picture} alt={firstname} size={60}/>
                    <ContentGrid>
                        <div>
                            <span>{`${title ? `${title}. ` : ''}${lastname} ${firstname}`}</span>
                        </div>
                        <div>
                            <span>{email}</span>
                        </div>
                    </ContentGrid>
                </Grid>
            </StyledWrapper>
        </Link>
    )
}

export default Tile
