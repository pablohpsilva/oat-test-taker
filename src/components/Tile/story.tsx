import React from 'react'
import { storiesOf } from '@storybook/react'
import { boolean, text, select, number, object } from '@storybook/addon-knobs'

import { Tile } from '@/components'
import uuid from '@/utils/uuid'
import readme from './readme.raw.md'

const TileComp = ({
    userId = text('userId', uuid()),
    firstName = text('firstName', 'Figgaro'),
    lastName = text('lastName', 'Delgado'),
    title = text('title', 'Mr'),
    picture = text('picture', 'https://placeimg.com/600/300/any')
}) => <Tile userId={userId} firstName={firstName} lastName={lastName} title={title} picture={picture} />

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|Tile', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <TileComp />, opts)
