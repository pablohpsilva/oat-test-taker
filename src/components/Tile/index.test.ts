import React from 'react'
import Tile from './'

test('should render the Tile component', async () => {
    const wrapper = shallow(<Tile userId="somerandomidhere" />)

    expect(wrapper.first().hasClass('Tile')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
