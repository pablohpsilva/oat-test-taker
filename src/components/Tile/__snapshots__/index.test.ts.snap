// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Tile component 1`] = `false`;

exports[`should render the Tile component 2`] = `
<Link
  to="/user/somerandomidhere"
>
  <Tile__StyledWrapper>
    <Tile__Grid>
      <Avatar
        size={60}
      />
      <Tile__ContentGrid>
        <div>
          <span>
            undefined undefined
          </span>
        </div>
        <div>
          <span />
        </div>
      </Tile__ContentGrid>
    </Tile__Grid>
  </Tile__StyledWrapper>
</Link>
`;
